docs/challenges/language-support/cpp.mdx

```diff
@@ -0,0 +1,93 @@
+---
+title: "C++"
+description: "Learn how to solve CodeCrafters challenges in C++"
+---
+
+# Local development
+
+To run the code for your challenge locally, you'll need `cmake` & `make` installed.
+
+The script for your challenge (like `./your_sqlite3.sh`) will automatically compile your code using the rules defined in `CMakeLists.txt` before executing it. It'll
+look something like this:
+
+```sh
+#!/bin/sh
+
+set -e
+cmake $(dirname $0) > /dev/null
+make > /dev/null
+exec ./server "$@"
+```
+
+# File structure
+
+- The files for your solution are placed in the `src/` directory.
+- `src/Server.cpp` contains the `main` function, which is what the test runner executes.
+- `CMakeLists.txt` contains the rules for compiling your program.
+
+# Adding more files
+
+You can add more files and directories to `src/`. The test runner will include them when compiling your code.
+
+For example, if you added a file at `src/Foo.hpp`, you could use it like so:
+
+<CodeGroup>
+```cpp src/Foo.hpp
+#pragma once
+
+#include <iostream>
+
+void foo() {
+std::cout << "Hello, World!";
+}
+
+````
+
+```cpp src/Server.cpp
+#include "Foo.hpp"
+
+int main(int argc, char* argv[]) {
+    foo()   // calls foo() defined in src/Foo.hpp
+}
+````
+
+</CodeGroup>
+
+<Note>
+  `#pragma once` is a preprocessor directive that ensures the header file is
```

Comment:
Is this really important? What happens if a user doesn't include it?

Reply:
@rohitpaulk It could lead to the header file being included multiple times which can lead to redefinition errors and greater compile times